cpp/src/DataStorm/SessionI.h

```diff
@@ -233,82 +241,82 @@ namespace DataStormI
         };
 
     public:
-        SessionI(const std::shared_ptr<NodeI>&, const std::shared_ptr<DataStormContract::NodePrx>&);
-        void init(const std::shared_ptr<DataStormContract::SessionPrx>&);
+        SessionI(const std::shared_ptr<NodeI>&, std::optional<DataStormContract::NodePrx>);
+        void init(std::optional<DataStormContract::SessionPrx>);
 
         virtual void announceTopics(DataStormContract::TopicInfoSeq, bool, const Ice::Current&) override;
         virtual void attachTopic(DataStormContract::TopicSpec, const Ice::Current&) override;
-        virtual void detachTopic(long long int, const Ice::Current&) override;
+        virtual void detachTopic(std::int64_t, const Ice::Current&) override;
 
-        virtual void attachTags(long long int, DataStormContract::ElementInfoSeq, bool, const Ice::Current&) override;
-        virtual void detachTags(long long int, DataStormContract::LongSeq, const Ice::Current&) override;
+        virtual void attachTags(std::int64_t, DataStormContract::ElementInfoSeq, bool, const Ice::Current&) override;
```

Comment:
In a follow-up, we should remove all these redundant virtual. A virtual function should have exactly one of "virtual", "override" or "final".